### PR TITLE
Textual Cleanup of the OpenQASM 3.0 Specification

### PR DESCRIPTION
**Pull request by jwoehr**
- Grammar, Syntax, and Usage
- Style (less purple)
- Clarity

### CODE REVIEW

File: source/intro.rst [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-a low-level pulse representation.
+OpenQASM is an imperative programming language that describes universal quantum
+computing using the measurement-based quantum circuit model. Its use is
+near-term quantum computing experiments.
+
+OpenQASM presents a parameterized set of physical logic gates and concurrent
+real-time classical computations. Its main goal is to serve as an intermediate
+representation for higher-level compilers to communicate with quantum hardware.
+Allowances have been made for human usability. In particular, the language admits
+different representations of the same program as it is descends from a high-level
```
Comment (taalexander) [inline source/intro.rst:12]:
```suggestion
different representations of the same program as it is transformed from a high-level
```

File: source/intro.rst [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+description to pulse representation.
```
Comment (taalexander) [inline source/intro.rst:13]:
```suggestion
description to a pulse representation.
```

File: source/language/comments.rst [hunk 3]
```diff
@@ -18,17 +18,23 @@ A comment block begins with ``/*`` and ends with ``*/``:
...
+``OPENQASM`` *M.m* ``;`` indicating a major version *M* and minor version *m*.
+
+This document describes Version 3.0.
+
+The minor version number, expressed as a decimal point after the major version
+number followed by a number, is optional. If not present, the minor version
+number is assumed to be zero.
+
+Multiple occurrences of the version string (outside of comments) are not
+permitted, nor may it appear elsewhere than as the first non-comment line.
```
Comment (taalexander) [inline source/language/comments.rst:31]:
```suggestion
permitted, nor may it appear elsewhere than the first non-comment line.
```

**Review by taalexander — CHANGES_REQUESTED** [review]
Thank you for these changes, I've attached a couple suggestions.

**Comment by jwoehr** [comment]
Hmm I've been looking at
- **Aliasing** language/types.html#aliasing
- **Register concatenation and slicing** language/types.html#register-concatenation-and-slicing

and I'm not sure they are consistent with each other.

In a `qubit name[5]` is the whole register `name[0:4]` or `name[0:5]`?

In any case, I will need to correct or enhance my last checking referencing slicing, which, though clearer than other explanation, is probably wrong :)

**Comment by jwoehr** [comment]
I've corrected what I wrote earlier so it's consistent with the section on slicing, but probably the discussion of slicing should be unified and made a bit clearer.

**Comment by jwoehr** [comment]
As I read it, the introduction of discussion of register slices in language/types#qubits was superfluous and the description of slicing in language/types#register-concatenation-and-slicing is entirely sufficient, so I have removed the superfluity.

**Review by taalexander — APPROVED** [review]

**Comment by taalexander** [comment]
Thank you very much for these contributions, it reads much clearer, and several errors we're fixed. We welcome many more 😄 .

**Pull request merged**